### PR TITLE
[SPARK-45310][CORE] Report shuffle block status should respect shuffle service during decommission migration

### PR DESCRIPTION
**Pull request by Ngone51**
### What changes were proposed in this pull request?
Currently, when migrating a shuffle block during decommission, its location type would be changed from shuffle service (if enabled) to the executor. This is of course not the expected behavior. We should still use the external shuffle service to fetch the migrated shuffle blocks. This PR fixes this issue by respecting the shuffle service when reporting the shuffle blocks.

### Why are the changes needed?

Bug fix/improvment.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

Unit test.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala [hunk 1]
```diff
@@ -133,12 +133,12 @@ class BlockManagerSuite extends SparkFunSuite with Matchers with PrivateMethodTe
     val bmSecurityMgr = new SecurityManager(bmConf, encryptionKey)
     val serializerManager = new SerializerManager(serializer, bmConf, encryptionKey)
     val transfer = transferService.getOrElse(new NettyBlockTransferService(
-      conf, securityMgr, serializerManager, "localhost", "localhost", 0, 1))
+      bmConf, securityMgr, serializerManager, "localhost", "localhost", 0, 1))
```
Comment (dongjoon-hyun) [inline core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala:136]: Is this an improvement to add a new test case or a bug, @Ngone51 ?

File: core/src/main/scala/org/apache/spark/storage/BlockManager.scala [hunk 2]
```diff
@@ -527,7 +527,9 @@ private[spark] class BlockManager(
       logInfo(s"external shuffle service port = $externalShuffleServicePort")
       shuffleServerId = BlockManagerId(executorId, blockTransferService.hostName,
         externalShuffleServicePort)
-      if (!isDriver) {
+      // Do not try to register with the external shuffle server under testing since none of
+      // test modes (local, local-cluster) enables shuffle service.
+      if (!isDriver && !Utils.isTesting) {
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/storage/BlockManager.scala:532]: This looks orthogonal. Shall we make a new JIRA and merge independently?

Comment (Ngone51) [reply core/src/main/scala/org/apache/spark/storage/BlockManager.scala:532]: Without this, the test can't run due to the `BlockManager` creation will fail on this.

File: core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala (hunk 1 above)
Comment (Ngone51) [reply core/src/test/scala/org/apache/spark/storage/BlockManagerSuite.scala:136]: It's a bug. `makeBlockManager` should use `testConf` instead of global `conf` when `testConf` exists.

File: core/src/main/scala/org/apache/spark/storage/BlockManager.scala (hunk 2 above)
Comment (Ngone51) [reply core/src/main/scala/org/apache/spark/storage/BlockManager.scala]: It turns out some tests rely on this creation failure for testing purposes. I added a separate testing conf (`spark.testing.skipESSRegister`) to keep the default behavior.

**Comment by Ngone51** [comment]
@dongjoon-hyun @holdenk Please take a look, thanks!

**Comment by dongjoon-hyun** [comment]
Thank you for pinging me. I will take a look tonight, @Ngone51 .

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @Ngone51 .

**Comment by Ngone51** [comment]
Thanks @dongjoon-hyun
